swarm-production/docker-stack.yml

```diff
@@ -76,27 +79,73 @@ services:
       - "80:80"
     environment:
       LISTEN_PORT: 80
-      UPSTREAM_SERVER: "s3-front:8000"
+      UPSTREAM_SERVER: "cloudserver-front:8000"
     networks:
       - frontend
       - frontend-dmz
     depends_on:
-      - s3-front
+      - cloudserver-front
     deploy:
       mode: global
       update_config:
         parallelism: 1
         delay: "10s"
         monitor: "5s"
 
+  zookeeper:
+    image: zookeeper
+    ports:
+      - "2181"
+    networks:
+      - backend
+    volumes:
+      - "zookeeper-datalog:/datalog:rw"
+      - "zookeeper-data:/data:rw"
+    deploy:
+      mode: replicated
+      placement:
+        constraints:
+          - node.labels.io.zenko.type == storage
+
+  kafka:
+    image: zenko/kafka
+    ports:
+      - "9092"
+    networks:
+      - backend
+    volumes:
+      - "kafka-data:/data:rw"
+    depends_on:
+      - zookeeper
+    deploy:
+      mode: replicated
+      placement:
+        constraints:
+          - node.labels.io.zenko.type == storage
+
+  backbeat:
+    image: zenko/backbeat
+    networks:
+      - backend
+    command: npm start
```

Comment:
is this supposed to start a task? how do we determine whether we are starting the replication task or the clueso task?

Reply:
I added the `start` rule in https://github.com/scality/backbeat/pull/88, which starts both the queue populator and the replication processor. We would need to selectively start the services that we'll need at processing stage, say replication/clueso/both, though I don't know what's the best way to deal with this at this point. Any idea?

Reply:
Note that on the populator side I assume there will be one populator for all services, with pluggable filters for each service (again this has to be properly designed)

Reply:
I like the idea of pluggable filters. Clueso has other tasks beyond the queue (creating the METADATA bucket for instance). Maybe there could be environment variables here for clueso or replication and if those are set, in backbeat we then start whatever processes beyond the populator are necessary?